src/megatron/energon/worker.py

```diff
@@ -218,5 +259,24 @@ def worker_log(self, data: dict) -> None:
             self._worker_debug_file.flush()
 
 
-#: The global worker config instance used if not overridden.
-global_worker_config = WorkerConfig.default_worker_config(num_workers=0)
+def set_global_seeds(seed: int):
+    """Sets the global seeds for torch, numpy and random."""
+    torch.manual_seed(seed)
+    numpy.random.seed(seed)
+    random.seed(seed)
+
+
+def save_global_rng_state() -> Dict[str, Any]:
+    """Saves the global rng state for torch, numpy and random."""
+    return {
+        "torch": torch.get_rng_state(),
+        "numpy": numpy.random.get_state(),
+        "random": random.getstate(),
+    }
+
+
+def restore_global_rng_state(state: Dict[str, Any]) -> None:
+    """Restores the global rng state for torch, numpy and random."""
+    torch.set_rng_state(state["torch"])
+    numpy.random.set_state(state["numpy"])
+    random.setstate(state["random"])
```

Comment:
Rather use a dataclass here? It's a dict with fixed keys.

Then, also make this methods?

```suggestion
@dataclass
class RandomState:
    torch: bytes # or whatever the type is
    numpy: bytes # or whatever the type is
    random: bytes # or whatever the type is

    @staticmethod
    def set_global_seeds(seed: int):
        """Sets the global seeds for torch, numpy and random."""
        torch.manual_seed(seed)
        numpy.random.seed(seed)
        random.seed(seed)

    @staticmethod
    def save_global_rng_state() -> 'RandomState':
        """Saves the global rng state for torch, numpy and random."""
        return RandomState(
            torch=torch.get_rng_state(),
            numpy=numpy.random.get_state(),
            random=random.getstate(),
        )

    def restore_global_rng_state(self) -> None:
        """Restores the global rng state for torch, numpy and random."""
        torch.set_rng_state(self.torch)
        numpy.random.set_state(self.numpy)
        random.setstate(self.random)
```

Reply:
Yeah we discussed this. I forgot about it. Will do